core/trino-main/src/test/java/io/trino/type/TestBigintType.java

```diff
@@ -16,6 +16,7 @@
 import io.trino.spi.block.Block;
 import io.trino.spi.block.BlockBuilder;
 import io.trino.spi.type.Type.Range;
+import org.junit.jupiter.api.Test;
```

Comment:
For me, these all run properly as is if you change the parent class `AbstractTestType` to use the `org.testng.annotations.Test` annotation instead of the junit `org.junit.jupiter.api.Test` annotation.

Reply:
Indeed, but that’s not how junit works — and we’re in the process of replacing testng with junit

Reply:
We should invest into writing a checker for this for the transition period like we have multiple for TestNG already. It seems very easy to do a migration but not realize multiple tests aren't running anymore.

Reply:
Yeah, this is a huge landmine. I already stepped on it some time ago and we briefly discussed countermeasures, but nothing came out of it. I suppose we could write an Error Prone check for this (I am surprised that there is no one already).

Reply:
> We should invest into writing a checker for this for the transition period like we have multiple for TestNG already. It seems very easy to do a migration but not realize multiple tests aren't running anymore.

😱 😱 😱 

BTW this risk was very well known even _before_ we did first junit baby steps. so it's the "landmine" we run onto deliberately. 

cc @losipiuk

Reply:
Requirement to put `@Test` on overridden test is just junit's "ux bug", or a quirk, and agreed that we definitely need a checker for that. Thanks @ebyhr for reporting https://github.com/trinodb/trino/issues/14027.